src/Castle.Core/DynamicProxy/Generators/GeneratorUtil.cs

```diff
@@ -74,25 +74,23 @@ bool IsReadOnly(ParameterInfo parameter)
 				//
 				// The above points inform the following detection logic: First, we rely on an IL
 				// `[in]` modifier being present. This is a "fast guard" against non-`in` parameters:
-				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) == ParameterAttributes.In)
-				{
-					// Here we perform the actual check. We don't rely on cmods because support
-					// for them is at current too unreliable in general, and because we wouldn't
-					// be saving much time anyway.
-					if (parameter.GetCustomAttributes(false).Any(IsIsReadOnlyAttribute))
-					{
-						return true;
-					}
-				}
+				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) != ParameterAttributes.In)
+					return false;
```

Comment:
I know it was me who originally added such a check, but I am no longer certain whether it is a terribly good idea to use `ParameterAttributes` as a fast guard against non-`in` parameters. The fact that `in` parameters are marked as `ParameterAttributes.In` isn't (AFAIK) [specified](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/readonly-ref.md#metadata-representation-of-in-parameters), after all, so perhaps we shouldn't rely on it...?

(I am not requesting a change. I just wanted to mention that I am having second thoughts here, in case anyone else feels the same way about this check.)

Reply:
Yep, I also noticed that. However, decided to keep as it significantly improves the performance. Let's remove it after we meet some real problem with that check 😊